core/pta/bcm/elog.c

```diff
@@ -89,12 +89,12 @@ static TEE_Result pta_elog_load_nitro_fw(uint32_t param_types,
 		return TEE_ERROR_ACCESS_DENIED;
 	}
 
-	src_vaddr = (vaddr_t)phys_to_virt((uintptr_t)src_paddr + offset,
-					  MEM_AREA_RAM_SEC);
-
 	buf = params[0].memref.buffer;
 	sz = params[0].memref.size;
 
+	src_vaddr = (vaddr_t)phys_to_virt((uintptr_t)src_paddr + offset,
+					  MEM_AREA_RAM_SEC, sz);
```

Comment:
If we make the effort to figure out the size of the memory area then I we should also check the return value of this function and not just crash in `memcpy()` if this fails.